tests/topotests/ospf_gr_helper/test_ospf_gr_helper.py

```diff
@@ -0,0 +1,1003 @@
+
+    step("Un Configure gr helper using the router id")
+    ospf_gr_r0 = {
+        "r0": {
+            "ospf": {
+                "graceful-restart": {
+                    "helper-only": ["1.1.1.1"],
+                    "opaque": True,
+                    "delete": True,
+                }
+            }
+        }
+    }
+    result = create_router_ospf(tgen, topo, ospf_gr_r0)
+    assert result is True, "Testcase {} : Failed \n Error: {}".format(tc_name, result)
+
+    step("Verify that GR helper router is disabled in the DUT for" " router id x.x.x.x")
+    input_dict = {"enabledRouterIds": [{"routerId": "1.1.1.1"}]}
+    dut = "r0"
+    result = verify_ospf_gr_helper(tgen, topo, dut, input_dict, expected=False)
+    assert (
+        result is not True
+    ), "Testcase {} : Failed, Helper role enabled for RR\n Error: {}".format(
+        tc_name, result
+    )
+    delete_ospf()
+    write_test_footer(tc_name)
+
+
+def test_ospf_gr_helper_tc2_p0(request):
+    """
+    OSPF GR on Broadcast : Verify DUT enters Helper mode when neighbor
+    sends grace lsa, helps RR to restart gracefully (RR = DR)
+    """
+    tc_name = request.node.name
+    write_test_header(tc_name)
+    tgen = get_topogen()
+
+    # Don't run this test if we have any failure.
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    global topo, intf, intf1, pkt
+
+    step("Bring up the base config as per the topology")
+    step(
+        "Configure DR priority as 99 in RR , DUT dr priority = 98 "
+        "& reset ospf process in all the routers"
+    )
+    reset_config_on_routers(tgen)
+    ospf_covergence = verify_ospf_neighbor(tgen, topo, lan=True)
+    assert (
+        ospf_covergence is True
+    ), "OSPF is not after reset config \n Error:" " {}".format(ospf_covergence)
+    ospf_gr_r0 = {
+        "r0": {"ospf": {"graceful-restart": {"helper-only": [], "opaque": True}}}
+    }
+    result = create_router_ospf(tgen, topo, ospf_gr_r0)
+    assert result is True, "Testcase {} : Failed \n Error: {}".format(tc_name, result)
+
+    ospf_gr_r1 = {
+        "r1": {"ospf": {"graceful-restart": {"helper-only": [], "opaque": True}}}
+    }
+    result = create_router_ospf(tgen, topo, ospf_gr_r1)
+    assert result is True, "Testcase {} : Failed \n Error: {}".format(tc_name, result)
+
+    step("Verify that DUT enters into helper mode.")
+
+    input_dict = {"activeRestarterCnt": 1}
+    gracelsa_sent = False
+    repeat = 0
+    dut = "r0"
+    while not gracelsa_sent and repeat < Iters:
+        gracelsa_sent = scapy_send_raw_packet(tgen, topo, "r1", intf1, pkt)
+        result = verify_ospf_gr_helper(tgen, topo, dut, input_dict)
+        if isinstance(result, str):
+            repeat += 1
+            gracelsa_sent = False
+
+    assert result is True, "Testcase {} : Failed \n Error: {}".format(tc_name, result)
+
+    delete_ospf()
+    write_test_footer(tc_name)
+
+
+def test_ospf_gr_helper_tc3_p1(request):
```

Comment:
This suffix is used to prioritise the test cases if we just pass p0, it will run priority p0 testcase. So this suit has p0 and p1 test cases, so no p2 cases.  I have fixed TCx numbering issue. Now it is clean and linear numbering.